app/controllers/api/v1/files_controller.rb

```diff
@@ -11,7 +11,7 @@ def show
 
       submission = solution.user == current_user ?
         solution.submissions.last :
-        solution.latest_iteration.submission
+        solution.latest_iteration&.submission
 
       content = submission.files.find_by(filename: params[:filepath])&.content if submission
```

Comment:
Will this not blow up here because submission will be `nil`?

Reply:
It won't, due to the `if submission` at the end.